src/core/etl/src/Flow/ETL/PHP/Type/Logical/MapType.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types=1);
+
+namespace Flow\ETL\PHP\Type\Logical;
+
+use Flow\ETL\PHP\Type\Logical\Map\MapKey;
+use Flow\ETL\PHP\Type\Logical\Map\MapValue;
+use Flow\ETL\PHP\Type\Type;
+use Flow\Serializer\Serializable;
+
+/**
+ * @implements Serializable<array{key: MapKey, value: MapValue}>
+ */
+final class MapType implements LogicalType, Serializable
```

Comment:
We need to add unit tests for those types, there is not much logic here, but lack of coverage will reduce mutation score

Reply:
Covered, for other types I will make a new PR.